File: plugins/vpn_plugin/android/src/main/kotlin/com/example/vpn_plugin/VpnPlugin.kt
package com.example.vpn_plugin


import io.flutter.embedding.engine.plugins.FlutterPlugin

/** VpnPlugin */
class VpnPlugin: FlutterPlugin{

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    PlatformApi.setUp(flutterPluginBinding.binaryMessenger, PlatformApiImpl())
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

internal class PlatformApiImpl : PlatformApi {
  override fun getPlatformType(request: GetPlatformTypeRequest): GetPlatformTypeResponse {
    return GetPlatformTypeResponse(platformType = "Android")
  }
}

File: plugins/vpn_plugin/macos/Classes/VpnPlugin.swift
import Cocoa
import FlutterMacOS

public class VpnPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
      let messenger = registrar.messenger
      PlatformApiSetup.setUp(binaryMessenger: messenger, api: PlatformApiImpl())
  }
}

public class PlatformApiImpl: NSObject, PlatformApi {
    func getPlatformType(request: GetPlatformTypeRequest) throws -> GetPlatformTypeResponse {
        let response = GetPlatformTypeResponse(platformType: "MacOS")
        return response
    }
}
